File: trunk/server/src/main/java/org/ngame/protocol/battle/MatchResponse.java
package org.ngame.protocol.battle;

import lombok.Data;
import org.ngame.protocol.BaseResponse;
import org.ngame.protocol.annotation.Proto;
import org.ngame.protocol.domain.RoomInfo;

import com.baidu.bjf.remoting.protobuf.annotation.Protobuf;

/**
 *
 * @author beykery
 */
@Data
@Proto(value = -20, description = "匹配")
public class MatchResponse extends BaseResponse
{
   @Protobuf(description = "房间")
   private RoomInfo room;
}

File: trunk/unityClientSample/Assets/src/protocol/MatchResponse.cs
using ProtoBuf;
using System.Collections.Generic;
using System;

namespace protocol
{
    /// <summary>
    /// 匹配 协议:-20
    /// </summary>
	[Proto(value=-20,description="匹配")]
	[ProtoContract]
	public class MatchResponse
	{
        /// <summary>
        ///  房间
        /// </summary>
		[ProtoMember(3, IsRequired = false)]
		public RoomInfo room;
        /// <summary>
        ///  是否成功
        /// </summary>
		[ProtoMember(1, IsRequired = false)]
		public bool success;
        /// <summary>
        ///  错误消息
        /// </summary>
		[ProtoMember(2, IsRequired = false)]
		public string info;

	}
}